### PR TITLE
code for generating discount code report

### PR DESCRIPTION
**Pull request by Bjvanminnen**
A simple script that can be used to generate a report on the current state of our discount codes. Not 100% sure bin/oneoff is the right place for this, but not really sure of a better place either.

After pasting to google sheets, it looks something like this
image

If this were a higher volume feature, it could make sense to actually build a report view for the web, but I think given that we're expecting <1000 codes to be used, one off is probably ok.

### CODE REVIEW

File: bin/oneoff/populate_discount_codes.rb [hunk 1]
```diff
@@ -110,3 +110,31 @@ def create_discount_eligible_section(teacher, section_name = 'Discount Eligible
...
+      app.admin_set_status,
+      app.signature,
+      app.signed_at,
+      app.full_discount,
+      app.circuit_playground_discount_code.try(:code),
+      app.updated_at,
+      app.school_id,
+    ].join(',')
+  end
+  puts rows
```
Comment (islemaster) [inline bin/oneoff/populate_discount_codes.rb:139]: Just experimenting: I think you can do this using the CSV module's [shortcut interface](http://ruby-doc.org/stdlib-2.0.0/libdoc/csv/rdoc/CSV.html#class-CSV-label-Shortcut+Interface), but I'm not sure it's actually any better:

```rb
def discount_code_reporting
  CSV do |csv|
    csv << [
      # Column names
    ]
    CircuitPlaygroundDiscountApplication.all.each do |app|
      csv << [
        # columns for row
      ]
    end
  end
end
```

File: bin/oneoff/populate_discount_codes.rb [hunk 2]
```diff
@@ -110,3 +110,31 @@ def create_discount_eligible_section(teacher, section_name = 'Discount Eligible
   end
   section
 end
+
+# This will generate a report on the status of all of our discount applications.
+# It will print out a bunch of rows, one per line. You can then copy paste this
+# into google sheets. After pasting, click on the little icon that shows up and
+# select "Split text into columns"
+def discount_code_reporting(filepath)
```
Comment (islemaster) [inline bin/oneoff/populate_discount_codes.rb]: Unused `filepath` argument?

**Comment by Bjvanminnen** [comment]
I had actually originally written this doing CSV. However, because it's being run on production-console, getting the file off of the machine and into google sheets becomes a bit of a pain. It felt easier to be able to just copy/paste from the console.

**Comment by islemaster** [comment]
The 'shortcut interface' I used there prints to stdout by default, so should be functionally equivalent to what you wrote. Of course, what you've got works perfectly well and has fewer dependencies. :grin:

**Pull request merged**